### PR TITLE
Moving remaining beacon tests to pytest

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Moving remaining beacon tests to pytest.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by dwoz — DISMISSED** [review]

File: tests/pytests/unit/beacons/test_salt_proxy.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+from collections import namedtuple
+
+import pytest
+
+# Salt libs
+import salt.beacons.salt_proxy as salt_proxy
+from tests.support.mock import MagicMock, patch
+
+PATCH_OPTS = dict(autospec=True, spec_set=True)
```
Comment (Ch3LL) [inline tests/pytests/unit/beacons/test_salt_proxy.py:11]: I'm not seeing `PATCH_OPTS` or `FakeProcess` being used anywhere. We also don't want to be using mutable objects at the global level.

File: tests/pytests/unit/beacons/test_service.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+from collections import namedtuple
+
+import pytest
+
+# Salt libs
+import salt.beacons.service as service_beacon
+from tests.support.mock import MagicMock, patch
+
+PATCH_OPTS = dict(autospec=True, spec_set=True)
```
Comment (Ch3LL) [inline tests/pytests/unit/beacons/test_service.py:11]: I'm not seeing `PATCH_OPTS` or `FakeProcess` being used anywhere. We also don't want to be using mutable objects at the global level.

File: tests/pytests/unit/beacons/test_smartos_imgadm.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
+# Python libs
+import pytest
+
+# Salt libs
+import salt.beacons.smartos_imgadm as imgadm
+from tests.support.mock import MagicMock, patch
+
+# Mock Results
+MOCK_CLEAN_STATE = {"first_run": True, "images": []}
```
Comment (Ch3LL) [inline tests/pytests/unit/beacons/test_smartos_imgadm.py:9]: we shouldn't be using mutable objects at the global level ( https://docs.saltproject.io/en/master/topics/development/tests/index.html#module-global-level-variables)

This includes this dictionary and the ones below.

File: tests/pytests/unit/beacons/test_smartos_vmadm.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
+# Python libs
+import pytest
+
+# Salt libs
+import salt.beacons.smartos_vmadm as vmadm
+from tests.support.mock import MagicMock, patch
+
+# Mock Results
+MOCK_CLEAN_STATE = {"first_run": True, "vms": []}
```
Comment (Ch3LL) [inline tests/pytests/unit/beacons/test_smartos_vmadm.py:9]: we shouldn't be using mutable objects at the global level ( https://docs.saltproject.io/en/master/topics/development/tests/index.html#module-global-level-variables)

This includes this dictionary and the ones below.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
looks like the telegram and twilio tests are being skipped (https://jenkins.saltproject.io/job/pr-centos-7-x86_64-py3-m2crypto-pytest/job/PR-60687/3/testReport/tests.pytests.unit.beacons/test_telegram_bot_msg/)

https://jenkins.saltproject.io/job/pr-centos-7-x86_64-py3-m2crypto-pytest/job/PR-60687/3/testReport/tests.pytests.unit.beacons/test_twilio_txt_msg/

should be able to add telegram and twilio to the requirements file.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**